### ISSUE TITLE
MixedAssignment false positive

### ISSUE THREAD

**Issue by chapa** [vimeo/psalm#4250]
As described in the [documentation](https://psalm.dev/docs/running_psalm/issues/MixedAssignment/), it is possible to fix MixedAssignment with :

```php
<?php

$a = $_GET['foo'];
assert(is_string($a));
```

It would be nice to be able to replace `assert(<test>)` by `if (!<test>) throw` : https://psalm.dev/r/bb5373b009

**Comment by ptomulik** [comment]
Do I have the same issue? Two functions, test1() and test2() should give same results, but test1() issues the error

https://psalm.dev/r/7b3de16944

**Comment by muglug** [comment]
Fixed in 549b200aa
